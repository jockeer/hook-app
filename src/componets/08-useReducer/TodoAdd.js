import React from 'react'
import useForm from '../../hooks/useForm'

export const TodoAdd = ({handleAddTodo}) => {

    const [ {desc}, handleInputChange, reset ] = useForm({
        desc:''
    })  

    const handleSubmit = e =>{
        e.preventDefault()
        if (desc.trim().length <= 1) {
            return
        }

        const newTodo = {
            id: new Date().getTime(),
            desc,
            done: false
        }
    
        handleAddTodo(newTodo)
        reset()
    }
    return (
        <>
            <h4>Agregar TODO</h4>
            <hr />
            <form onSubmit={handleSubmit}>
                <input 
                    type="text" 
                    name='desc'
                    placeholder='Aprender...'
                    autoComplete='off'
                    className='form-control'
                    value={desc}
                    onChange={handleInputChange}
                />

                <div className='d-grid gap-2'>
                    <button type='submit' className="btn btn-outline-primary btn-block mt-3">
                        Agregar
                    </button>
                
                </div>
            </form>
        </>
    )
}
