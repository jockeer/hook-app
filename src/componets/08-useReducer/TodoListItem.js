import React from 'react';

const TodoListItem = ({todo, i, handleToggle, handleDelete}) => {
    return (  
        <li 
            className="list-group-item">
            <p className={`${ todo.done && 'complete' }`} onClick={()=>handleToggle(todo.id)}>{i+1}.- {todo.desc}</p> 
            <button type='button' className="btn btn-danger" onClick={ () => handleDelete(todo.id)}>Borrar</button>
        </li>
    );
}
 
export default TodoListItem;