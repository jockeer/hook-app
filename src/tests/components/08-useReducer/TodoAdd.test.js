import React from 'react';

import {shallow} from 'enzyme'
import { TodoAdd } from '../../../componets/08-useReducer/TodoAdd';

describe('Prueba TodoAdd', () => {  

    const handleAddTodo = jest.fn()
    const wrapper = shallow(<TodoAdd handleAddTodo={handleAddTodo}/>)

    test('Debe mostrarse correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot()
        
    })

    test('NO debe llamar handleAddTodo', () => {  

        const formSubmit = wrapper.find('form').prop('onSubmit')
        
        formSubmit({preventDefault(){}})
        
        expect(handleAddTodo).toHaveBeenCalledTimes(0)
        
    })
    
    test('Debe llamar la funcion handleAddTodo', () => { 
        
        const value= 'Aprender MongoDB'
        const input = wrapper.find('input')
        input.simulate('change', { 
            target:{ 
                value, 
                name:'desc'
            } 
        })
        
        const formSubmit = wrapper.find('form').prop('onSubmit')
        
        formSubmit({preventDefault(){}})

        expect(handleAddTodo).toHaveBeenCalledTimes(1)
        expect(handleAddTodo).toHaveBeenCalledWith(expect.any(Object))
        expect(handleAddTodo).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc:value,
            done:false
        })

        expect(wrapper.find('input').prop('value')).toBe('')


    })
    

})