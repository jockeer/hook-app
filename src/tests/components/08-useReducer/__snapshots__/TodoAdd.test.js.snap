// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba TodoAdd Debe mostrarse correctamente 1`] = `
<Fragment>
  <h4>
    Agregar TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="desc"
      onChange={[Function]}
      placeholder="Aprender..."
      type="text"
      value=""
    />
    <div
      className="d-grid gap-2"
    >
      <button
        className="btn btn-outline-primary btn-block mt-3"
        type="submit"
      >
        Agregar
      </button>
    </div>
  </form>
</Fragment>
`;
